Training25/Program.cs

```diff
@@ -3,11 +3,226 @@
+      WriteLine ("\nTest with string");
+      List<string> str3 = ["xyz", "abc"];
+      MyList<string> str4 = new ();
+      str4.Add ("xyz");
+      str4.Add ("abc");
+      WriteLine ("Default List<string>:");
+      PrintList (str3);
+      WriteLine ("MyList<string>:");
+      str4.Print ();
+
+      // Helper method for default List<T> to print the elements in the list
+      static void PrintList<T> (List<T> list) {
+         foreach (T item in list) Write ($"{item} ");
+         WriteLine ();
+      }
    }
 }
+#endregion
+
+#region class MyList<T> ---------------------------------------------------------------------------
+class MyList<T> {
+
+   #region Constructor ----------------------------------------------
+   /// <summary>Constructor that initializes a new list with the default capacity</summary>
+   public MyList () {
+      mArray = new T[DEFCAP];
+      mCnt = 0;
+   }
+   #endregion
+
+   #region Properties -----------------------------------------------
+   /// <summary>Gets the number of elements currently stored in the list</summary>
+   public int Count => mCnt;
+
+   /// <summary>Gets the current capacity of the list</summary>
+   public int Capacity => mArray.Length;
+
+   /// <summary>Gets or sets the element at the specified index</summary>
+   public T this[int index] {
+      get { // executes when a value is read
+         if (index < 0 || index >= mCnt) throw new IndexOutOfRangeException ();
+         return mArray[index];
+      }
+      set { // executes when a value is written
+         if (index < 0 || index >= mCnt) throw new IndexOutOfRangeException ();
+         mArray[index] = value;
+      }
+   }
+   #endregion
+
+   #region Methods --------------------------------------------------
+   /// <summary>Adds the element at the end of the list</summary>
+   public void Add (T a) {
+      CheckCapacity ();
+      mArray[mCnt++] = a;
+   }
+
+   /// <summary>Removes the first occurrence of the given element</summary>
+   public bool Remove (T a) {
+      int idx = IndexOf (a);
+      if (idx == -1) return false;
+      RemoveAt (idx);
+      return true;
+   }
+
+   /// <summary>Clears all stored elements while keeping the current capacity unchanged</summary>
+   public void Clear () {
+      Array.Clear (mArray, 0, mCnt);
+      mCnt = 0;
+   }
+
+   /// <summary>Inserts the element at the specified index</summary>
+   public void Insert (int index, T a) {
+      if (index < 0 || index > mCnt) throw new ArgumentOutOfRangeException (nameof (index));
+      CheckCapacity ();
+      Array.Copy (mArray, index, mArray, index + 1, mCnt - index);
+      mArray[index] = a;
+      mCnt++;
+   }
+
+   /// <summary>Removes the element at the specified index</summary>
+   public void RemoveAt (int index) {
+      if (index < 0 || index >= mCnt) throw new ArgumentOutOfRangeException (nameof (index));
+      Array.Copy (mArray, index + 1, mArray, index, mCnt - index - 1);
+      mCnt--;
+      mArray[mCnt] = default!;

```

Comment:
It can be simplified as
```suggestion 
      mArray[mCnt--] = default!;
```